src/Features/Core/Portable/NavigateTo/NavigateToSearchResultComparer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.NavigateTo
+{
+    /// <summary>
+    /// Comparer that considers to navigate to results the same if they will navigate to the same document and span.
+    /// This ensures that we don't see tons of results for the same symbol when a file is linked into many projects.
+    /// </summary>
+    internal class NavigateToSearchResultComparer : IEqualityComparer<INavigateToSearchResult>
+    {
+        public static readonly IEqualityComparer<INavigateToSearchResult> Instance = new NavigateToSearchResultComparer();
+
+        private NavigateToSearchResultComparer()
+        {
+        }
+
+        public bool Equals(INavigateToSearchResult? x, INavigateToSearchResult? y)
+            => x?.NavigableItem.Document.FilePath == y?.NavigableItem.Document.FilePath &&
+               x?.NavigableItem.SourceSpan == y?.NavigableItem.SourceSpan;
```

Comment:
Given the goal of this PR is to dedupe results across project flavors, do we want to take into account the project name from NameAndFlavor here? Otherwise, we will lose the info that a symbol linked into multiple projects, by which I mean truly independent projects, not the result of multi-targeting. (Although I can see argument for showing fewer results from linked documents, since they all navigate to the same doc)

Reply:
i actually think this logic is the most desirable.  showing multiple results taht just all navigate to essentially the exact same file/loc is just noise for the user.

Reply:
Definitely see the benefit of this approach. I think it's a worthy trade-off for the loss of info in limited scenarios :)
Could you please call this out in the comments, so it's clear it is intended?

Reply:
sure.

Reply:
Done.